pkg/apiserver/profiling/model.go

```diff
@@ -49,7 +63,8 @@ func (TaskModel) TableName() string {
 
 type TaskGroupModel struct {
 	ID                  uint                          `json:"id" gorm:"primary_key"`
-	State               TaskState                     `json:"state" gorm:"index"`
+	State               TaskState                     `json:"state" gorm:"index"` // legacy state, works before version 5.3.0
```

Comment:
This can be safely changed to:

```suggestion
	State               GroupTaskState                     `json:"state" gorm:"index"` // legacy state, works before version 5.3.0
```

as long as you introduce `GroupTaskStateFinished = 2`.

BTW there is no "version 5.3.0" concept in TiDB Dashboard. Please use TiDB Dashboard's release version.